include/seqan3/core/algorithm/configuration.hpp

```diff
@@ -572,14 +570,11 @@ class configuration : public std::tuple<configs_t...>
 /*!\brief Deduces the correct configuration element type from the passed seqan3::pipeable_config_element.
  * \relates seqan3::configuration
  */
-template <typename derived_t, typename value_t>
-configuration(pipeable_config_element<derived_t, value_t> &&) -> configuration<std::remove_cvref_t<derived_t>>;
-
-/*!\brief Deduces the correct configuration element type from the passed seqan3::pipeable_config_element.
- * \relates seqan3::configuration
- */
-template <typename derived_t, typename value_t>
-configuration(pipeable_config_element<derived_t, value_t> const &) -> configuration<std::remove_cvref_t<derived_t>>;
+template <typename config_t>
+//!\cond
+    requires detail::config_element_specialisation<std::remove_cvref_t<config_t>>
+//!\endcond
+configuration(config_t &&) -> configuration<std::remove_cvref_t<config_t>>;
```

Comment:
Do deduction guides need constraints? :thinking:

Reply:
They are hypothetical function templates which also participate in overload resolution. There are some rules about when a deduction guide is preferred over a regular constructor, but this is not important here. But I can remove the clutter by just taking the arugment by value such that I don't have to explicitly decay the type.

Reply:
Interesting! Thanks for the explanation.